crates/common/tedge_config/src/tedge_config.rs

```diff
@@ -305,6 +305,27 @@ impl ConfigSettingAccessor<MqttPortSetting> for TEdgeConfig {
     }
 }
 
+impl ConfigSettingAccessor<HttpPortSetting> for TEdgeConfig {
+    fn query(&self, _setting: HttpPortSetting) -> ConfigSettingResult<Port> {
+        Ok(self
+            .data
+            .mqtt
```

Comment:
This is the issue with `tedge_config`  ;-) Easy to add a new config parameter, but also easy to get it wrong!

```suggestion
            .http
```